core/src/main/java/io/questdb/griffin/FunctionFactoryDescriptor.java

```diff
@@ -230,7 +236,7 @@ public static String replaceSignatureName(String name, String signature) throws
     public int getArgTypeMask(int index) {
         int arrayIndex = index / 2;
         long mask = argTypes[arrayIndex];
-        return (int) ((mask >>> (32 - (index % 2) * 32)));
+        return (int) (mask >>> (32 - (index % 2) * 32));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).